### DISCUSSION TITLE
unexpected effects of disabled but connected nodes?

### DISCUSSION THREAD

**Discussion by edmundchong** [bonsai-rx/bonsai#2006] [Q&A] [answered]
I created a frequently updated `BehaviorSubject` which I pass through some conditional statements. However, there are strange effects from nodes that are commented out (but still connected). Here is a toy example which isolates the issue:

<img width="485" alt="image" src="https://github.com/user-attachments/assets/0db58b86-3d00-47fb-b2a7-915837729481">

In this case mySubject is only initialized once but is propagated 3 times.

When the commented out node is deleted or simply connected on the other end like the rest of the `Condition` nodes, everything is fine:
<img width="488" alt="image" src="https://github.com/user-attachments/assets/a6f19bd3-3701-405f-8477-15e36bf8c21c">.

The same behavior happens also if I e.g. replace `Merge` with `Zip` or `CombineLatest` or `Amb`.

Workflow attached below.

[foo.bonsai.zip](https://github.com/user-attachments/files/16922950/foo.bonsai.zip)

**Comment by bruno-f-cruz** [comment]
@glopesdev seems like a bug where the expression tree is seeing the disabled node as a hanging branch and adding a publish. Can you confirm? I am tentatively opening an issue with this. Thanks for reporting!

**Reply by glopesdev** [reply, marked as answer]
Yes, this is a known "quirk". There are is an intrinsic ambiguity in fully disabled branches between whether the source should be connected directly to the merge, or the branch removed entirely. A dangling branch is ultimately still merged back into the workflow output so it falls into this ambiguity too.

It is awkward though so raising the issue is fine and we can at least track the discussion on the impact of different ways to resolve it.
